### PR TITLE
Fix top-level alpha command help output and updates docs for v0.13.0 release

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Updates the HideMaturity method to also consider self such that any
nodes that hide their children will also hide themselves if they do not
match the specified maturity level.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

Cleans up workflow documentation by removing unsupported functionality.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

Updates the commands doc to reflect that state of commands for v0.13.0.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

Adds documentation describing the up command maturity framework and how
commands are migrated to one level of maturity to another.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

Fixes https://github.com/upbound/up/issues/237

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

Before:

```
🤖 (up) up upbound -h
Usage: up upbound <command>

Interact with Upbound.

Flags:
  -h, --help                          Show context-sensitive help.
  -v, --version                       Print version and exit.
  -q, --quiet                         Suppress all output.
      --pretty                        Pretty print output.

      --kubeconfig=STRING             Override default kubeconfig path.
  -n, --namespace="upbound-system"    Kubernetes namespace for Upbound ($UPBOUND_NAMESPACE).

Commands:
🤖 (up) up alpha upbound -h
Usage: up alpha upbound <command>

Interact with Upbound.

Flags:
  -h, --help                          Show context-sensitive help.
  -v, --version                       Print version and exit.
  -q, --quiet                         Suppress all output.
      --pretty                        Pretty print output.

      --kubeconfig=STRING             Override default kubeconfig path.
  -n, --namespace="upbound-system"    Kubernetes namespace for Upbound ($UPBOUND_NAMESPACE).

Commands:
  alpha upbound install <version>
    Install Upbound.

  alpha upbound mail
    Run a local mail portal.

  alpha upbound uninstall [<name>]
    Uninstall Upbound.

  alpha upbound upgrade <version>
    Upgrade Upbound.
```

After:
```
🤖 (up) up upbound -h
Refusing to emit help for hidden command. See alpha variant.
🤖 (up) up alpha upbound -h
Usage: up alpha upbound <command>

Interact with Upbound.

Flags:
  -h, --help                          Show context-sensitive help.
  -v, --version                       Print version and exit.
  -q, --quiet                         Suppress all output.
      --pretty                        Pretty print output.

      --kubeconfig=STRING             Override default kubeconfig path.
  -n, --namespace="upbound-system"    Kubernetes namespace for Upbound ($UPBOUND_NAMESPACE).

Commands:
  alpha upbound install <version>
    Install Upbound.

  alpha upbound mail
    Run a local mail portal.

  alpha upbound uninstall [<name>]
    Uninstall Upbound.

  alpha upbound upgrade <version>
    Upgrade Upbound.
```

### CODE REVIEW

File: docs/commands.md [hunk 1]
```diff
@@ -64,109 +67,214 @@ Top-level flags can be passed for any top-level or group-specific command.
...
+        `kubectl` are used if not provided.
+        - `-n,--namespace = STRING` (Env: `UPBOUND_NAMESPACE`) (Default:
+          `upbound-system`): Kubernetes namespace used for installing and
+          managing Upbound.
+    - Behavior: Creates a package pull secret in the cluster pointed to by the
+      currently configured `kubeconfig`. If `-f` is not provided, the session
+      token in the current profile is used. Session tokens expire within 30
+      days. If `-f` is provided, it must point to a file with valid token
+      credentials in the following JSON format: `{"accessId": "<access-id>",
+      "token":"<token>"}`. This is the same format emitted by `up robot token
```
Comment (tnthornton) [inline docs/commands.md:113]: Thoughts on adjusting to:
```suggestion
      credentials in the following JSON format: 
      ```json
      {
        "accessId": "<access-id>",
        "token": "<token>"
      }
      ```
      This is the same format emitted by `up robot token
```

File: docs/commands.md [hunk 2]
```diff
@@ -248,52 +356,50 @@ may choose not to utilize the group flags when not relevant.
...
 Commands in the **XPKG** group are used to build, push, and interact with
-on-disk UXP packages.
+on-disk Crossplane packages.
 
 - `build`
     - Flags:
-        - `--name = STRING`: Name of the package to be built. Uses name in
-          crossplane.yaml if not specified. Does not correspond to package tag.
-        - `-f, --package-root = STRING`: Path to package directory.
+        - `--name = STRING` (*DEPRECATED: use `--output` instead.*): Name of the
```
Comment (tnthornton) [inline docs/commands.md:363]: Do we have a sunset policy for when we remove commands labeled `deprecated`? For example:
* deprecated 8/25
* removed 9/25

**Review by tnthornton — APPROVED** [review]
LGTM @hasheddan. Just a few comments, feel free to address if you want 👍.

File: docs/commands.md (hunk 1 above)
Comment (hasheddan) [reply docs/commands.md:113]: Much better! Will update 👍🏻

File: docs/commands.md (hunk 2 above)
Comment (hasheddan) [reply docs/commands.md:363]: Unfortunately not -- I would love to introduce this, but for the time-being I don't feel confident about actually imposing any strict policy :/

**Pull request merged**